### PR TITLE
feat: secrets in params

### PR DESCRIPTION
**Pull request by exu**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by vsukhin — APPROVED** [review]
looks good, a few minor notes, but if I understood it right, we don't track changes in secrets key values

File: internal/app/api/v1/tests.go [hunk 1]
```diff
@@ -56,6 +56,9 @@ func (s TestkubeAPI) GetTestWithExecutionHandler() fiber.Handler {
 		}
 
 		test := testsmapper.MapTestCRToAPI(*crTest)
+
+		fmt.Printf("%+v\n", test)
```
[the remark this answers is not in the document]
Comment (exu) [reply internal/app/api/v1/tests.go:60]: yes! :)

File: pkg/api/v1/testkube/model_variable_type_extended.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package testkube
+
+func VariableTypePtr(stepType VariableType) *VariableType {
+	return &stepType
+}
+
+var VariableTypeBasic = VariableTypePtr(BASIC_VariableType)
```
[the remark this answers is not in the document]
Comment (exu) [reply pkg/api/v1/testkube/model_variable_type_extended.go:7]: It's swagger-gen stuff :/ we should try this new one. It's why I've introduced this `VariableType....` to simplify rewrite in future.

File: docs/reference.md [hunk 3]
```diff
@@ -1 +0,0 @@
-[0;93mparsing client version - : Invalid Semantic Version[0m
```
[the remark this answers is not in the document]
Comment (exu) [reply docs/reference.md:1]: this reference file looks like some leftover. removed it.

**Comment by exu** [comment]
> looks good, a few minor notes, but if I understood it right, we don't track changes in secrets key values

yes! thats the idea, 
executions has corresponding secrets only when there are explicit secret values.

**Pull request merged**